analyzers/carthage/carthage.go

```diff
@@ -0,0 +1,172 @@
+	}
+
+	// Parse and validate options.
+	var options Options
+	err = mapstructure.Decode(m.Options, &options)
+	if err != nil {
+		return nil, err
+	}
+	log.Logger.Debug("Decoded options: %#v", options)
+
+	analyzer := Analyzer{
+		CarthageCmd:     cartCmd,
+		CarthageVersion: cartVersion,
+
+		Carthage: carthage.Carthage{
+			Bin: cartCmd,
+		},
+
+		Options: options,
+		Module:  m,
+	}
+
+	log.Logger.Debugf("%#v", analyzer)
+	return &analyzer, nil
+}
+
+// Discover constructs modules in all directories with a `Cartfile`.
+func Discover(dir string, options map[string]interface{}) ([]module.Module, error) {
+	var modules []module.Module
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			log.Logger.Debugf("Failed to access path %s: %s\n", path, err.Error())
+			return err
+		}
+
+		moduleName := filepath.Base(path)
+
+		if !info.IsDir() && (info.Name() == "Cartfile" || info.Name() == "Cartfile.private") {
+			// If the same directory has both a Cartfile and Cartfile.private, we will just pick the Cartfile
+			if info.Name() == "Cartfile.private" {
+				cartfilePath := filepath.Join(filepath.Dir(path), "Cartfile")
+				cartfileExists, err := files.Exists(cartfilePath)
+				if err != nil {
+					log.Logger.Debugf("Error searching for Cartfile: %s (%s)", path, err.Error())
+				}
+
+				if cartfileExists == true {
+					return nil // This will get picked up before/later if it exists in the same directory
+				}
+			}
+
+			log.Logger.Debugf("Found Carthage package: %s (%s)", path, moduleName)
+			relPath, _ := filepath.Rel(dir, path)
+			modules = append(modules, module.Module{
+				Name:        moduleName,
+				Type:        pkg.Carthage,
+				BuildTarget: filepath.Dir(relPath),
+				Dir:         filepath.Dir(relPath),
+			})
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("Could not find Carthage package manifests: %s", err.Error())
+	}
+
+	return modules, nil
+}
+
+// IsBuilt checks whether file `Cartfile.resolved` exists, and also if a `Carthage` folder exists
+func (a *Analyzer) IsBuilt() (bool, error) {
+	log.Logger.Debugf("Checking Carthage build: %#v", a.Module)
+	hasResolvedCartfile, err := files.Exists(filepath.Join(a.Module.Dir, "Cartfile.resolved"))
+
+	if err != nil {
+		log.Logger.Warningf("Error checking Carthage build: %#v", err.Error())
+		return false, err
+	}
+
+	hasCarthageFolder, err := files.ExistsFolder(a.Module.Dir, "Carthage")
+	if err != nil {
+		log.Logger.Warningf("Error checking Carthage build: %#v", err.Error())
+		return false, err
+	}
+
+	isBuilt := hasCarthageFolder && hasResolvedCartfile
+
+	log.Logger.Debugf("Done checking Carthage build: %#v", isBuilt)
+	return isBuilt, nil
+}
+
+func (a *Analyzer) Clean() error {
+	return files.Rm(a.Module.Dir, "Carthage")
+}
+
+func (a *Analyzer) Build() error {
+	return a.Carthage.Install(a.Module.Dir)
+}
+
+func (a *Analyzer) Analyze() (graph.Deps, error) {
+	var deps graph.Deps
```

Comment:
[nit] I don't think this is ever used except to return a zero value in line 144.